packages/core/useVirtualList/index.ts

```diff
@@ -16,10 +17,17 @@ export interface UseVirtualListOptions {
   overscan?: number
 }
 
-export function useVirtualList <T = any>(list: T[], options: UseVirtualListOptions) {
+export type UseVirtualListItem<T> = {
+  data: T
+  index: number
+}
+
+export function useVirtualList <T = any>(list: MaybeRef<T[]>, options: UseVirtualListOptions) {
   const containerRef: Ref = ref<HTMLElement | null>()
   const size = useElementSize(containerRef)
-  const currentList: Ref = ref([])
+
+  const currentList: Ref<UseVirtualListItem<T>[]> = ref([])
+  const _list = ref(isRef(list) ? list.value : list)
```

Comment:
I don't believe this line is correct and this is why there are type errors.

Reply:
I feel like I want to use the sync composable instead of creating a new ref like this.